src/Components/GameBoardContainer/GameBoardContainer.js

```diff
@@ -0,0 +1,53 @@
+import GameBoard from '../GameBoard/GameBoard';
+import Keyboard from '../Keyboard/Keyboard';
+import newGameButton from '../../assets/new-game-button.png';
+import './GameBoardContainer.css'
+
+const GameBoardContainer = ({ typeLetter, deleteLetter, typedLetters, currentWordInPlay, enterGuess, submittedWords, gameOver, error, startNewGame }) => {
+  const displayNewGameButton = () => {
+    return (
+      <div className='new-game-button-container'>
+        <img
+          className='new-game-button'
+          src={newGameButton}
+          alt="new game button"
+          onClick={() => startNewGame()}
+        />
+      </div>
+    );
+  }
+
+  const displayErrorMessage = () => {
+    return (
+      <p className='error-message'>
+        {error}
+      </p>
+    );
+  }
+
+  return (
+    currentWordInPlay !== null ?
+      <div className='game-board-container'>
+        <div className='game-board-header'>
+          {error ? displayErrorMessage() : null}
+          {gameOver ? displayNewGameButton() : null}
+        </div>
+        <GameBoard
+          typedLetters={typedLetters}
+          submittedWords={submittedWords}
+          currentWordInPlay={currentWordInPlay}
+        />
+        <Keyboard 
+          typeLetter={typeLetter}
+          deleteLetter={deleteLetter}
+          enterGuess={enterGuess}
+          submittedWords={submittedWords}
+          currentWordInPlay={currentWordInPlay}
+        />
+      </div>
+      :
+      <div>Loading...</div>
```

Comment:
![](https://mir-s3-cdn-cf.behance.net/project_modules/max_1200/b6e0b072897469.5bf6e79950d23.gif)

Reply:
I'll definitely add a loading component and get rid of the "Loading..." text. Thank you for reminding me Elise!